spec/pharos/phases/validate_hostname_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'pharos/phases/validate_hostname'
+
+describe Pharos::Phases::ValidateHostname do
+  let(:ssh) { instance_double(Pharos::SSH::Client) }
+  let(:config) { Pharos::Config.new(
+      hosts: [
+        Pharos::Configuration::Host.new(
+          address: '192.0.2.1',
+        ),
+        Pharos::Configuration::Host.new(
+          address: '192.0.2.2',
+        ),
+        Pharos::Configuration::Host.new(
+          address: '192.0.2.3',
+        ),
+      ],
+  ) }
+
+  subject { described_class.new(config.hosts[0], config: config, ssh: ssh) }
+
+
+  describe '#call' do
+    context 'no duplicate hostnames' do
+      it 'raises if no duplicates' do
+        config.hosts[0].hostname = "host-0"
+        config.hosts[1].hostname = "host-1"
+        config.hosts[2].hostname = "host-2"
+
+        subject.call
+      end
+    end
+
+    context 'duplicate hostnames' do
+      it 'raises if no duplicates' do
+        config.hosts[0].hostname = "foo"
+        config.hosts[1].hostname = "foo"
+        config.hosts[2].hostname = "foo"
+
+        expect{ subject.call }.to raise_error(Pharos::InvalidHostError, "Duplicate hostname foo for hosts 192.0.2.2,192.0.2.3")
```

Comment:
Hmm... shouldn't this also include `192.0.2.1`?

Reply:
IMO no. `192.0.2.1` is the host we're currently checking, so the list only shows other hosts that have same hostname with that.

Reply:
Hmm... but if running with `parallel: false`, the `PhaseManager` currently doesn't actually log what host failed, I think... maybe that's something that should be fixed up there. With `parallel: true`, the `PhaseManager` does log the error with the failing host before letting it get raisd.

Reply:
So the actual error output ATM looks like this, it doesn't say which two hosts had duplicate hostnames:

```
==> Validate unique hostnames for all hosts @ 167.99.36.141 167.99.36.158 167.99.36.223
    [167.99.36.141] Validating hostname uniqueness ...
    [167.99.36.158] Validating hostname uniqueness ...
Pharos::InvalidHostError : Duplicate hostname terom-kupo-node for hosts 167.99.36.223
```

But I'd consider that a `PhaseManager` bug.

Reply:
I'd interpret that as `167.99.36.158`  having same hostname with `167.99.36.223`. Is that not true in this case?

Reply:
Yes, you need to match up the error with the previous line... it's not obvious IMO, particularly since it's not the same for the parallel phase output/error case.

Reply:
For me it was obvious. :) Yes, the error reporting could be enhanced at `PhaseManager` level.